src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -1021,3 +978,92 @@ EnsureReplicationSettings(Oid relationId, char replicationModel)
 								"factor\" to one%s.", extraHint)));
 	}
 }
+
+
+/*
+ * CopyLocalData copies local data into the shards.
```

Comment:
This is a pretty ambiguous comment. _the shards_?

I feel like—in addition to a more descriptive function comment—this function would also benefit from comments breaking it into "sections", i.e. to denote the relevant "steps" in this algorithm, as the bookkeeping variables tend to muck up the vertical space quite badly.

Finally, if there's analogous code in PostgreSQL, might be good to point to it for inspiration.